### PR TITLE
Code coverage configuration

### PR DESCRIPTION
**Pull request by rpeshkov**
Added code coverage configuration.

There are 3 types of reports currently generated:

- Text
- HTML
- LCOV

Samples:

screen shot 2018-10-17 at 02 31 53

screen shot 2018-10-17 at 02 31 40

I didn't provide screenshot on how LCOV report look like, but it's very usable format that can be loaded into e.g. this extension https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters and you'll see code coverage in text editor gutters. Refer to readme of extension for details.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -260,9 +260,14 @@
...
         "mocha": "^2.3.3",
         "@types/node": "^6.0.40",
-        "@types/mocha": "^2.2.32"
+        "@types/mocha": "^2.2.32",
+        "@types/glob": "^7.1.1"
     },
     "dependencies": {
-        "date-fns": "^1.28.3"
+        "date-fns": "^1.28.3",
+        "decache": "^4.4.0",
```
Comment (AdrieanKhisbe) [inline package.json]: the four new dependencies should be devDependencies

**Review by AdrieanKhisbe — CHANGES_REQUESTED** [review]
Awesome!

some minor change to be made though

**Comment by AdrieanKhisbe** [comment]
@rpeshkov Do you think whether it's possible to extract most of test/index.ts logic into a library to minimise set up of coverage of other extension, and keep the index simple?

**Comment by rpeshkov** [comment]
> @rpeshkov Do you think whether it's possible to extract most of test/index.ts logic into a library to minimise set up of coverage of other extension, and keep the index simple?

I'm not sure, need to check this. I'll try to do this today in the evening.

**Comment by AdrieanKhisbe** [comment]
> > @rpeshkov Do you think whether it's possible to extract most of test/index.ts logic into a library to minimise set up of coverage of other extension, and keep the index simple?
> 
> I'm not sure, need to check this. I'll try to do this today in the evening.

There is no Hurry Roman, I was just evaluating the possibility as I have some personal vscode-extension. We won't delay merging of PR for this. 😃

**Comment by rpeshkov** [comment]
@AdrieanKhisbe test runner extracted https://www.npmjs.com/package/vscode-coverage-test-runner

Everything works pretty well, changes to `index.ts`, compared to what it was initially, are minimal. Basically, only import path was changed.

Current NPM package is a little bit dummy, because I didn't bother with extracting configuration, so some paths are hardcoded for now, but I'll definitely make some updates later.

**Comment by AdrieanKhisbe** [comment]
@rpeshkov Awesome work!

**Comment by AdrieanKhisbe** [comment]
@rpeshkov I forked it and configure it a bit: https://github.com/rpeshkov/vscode-coverage-test-runner/pull/1

**Pull request closed without merging**